File: src/test/java/leetcode_challenges/april/day29/LC124BinaryTreeMaximumPathSumTest.java
package leetcode_challenges.april.day29;

import org.junit.Test;

import static org.junit.Assert.assertEquals;

public class LC124BinaryTreeMaximumPathSumTest {
    @Test
    public void testFidMax() {
        LC124BinaryTreeMaximumPathSum instance = new LC124BinaryTreeMaximumPathSum();

        TreeNode root = new TreeNode(2);
        root.left = new TreeNode(1);
        root.right = new TreeNode(3);

        assertEquals(instance.maxPathSum(root), 6);
    }
}

File: src/test/scala/leetcode_challenges/april/LC124_BinaryTreeMaximumPathSumTest.scala
package leetcode_challenges.april

import leetcode_challenges.april.day29.LC124_BinaryTreeMaximumPathSum
import org.scalatest.FlatSpec

class LC124_BinaryTreeMaximumPathSumTest extends FlatSpec{

  "LC124_BinaryTreeMaximumPathSum" should "find the maximum path sum of given non-empty binary tree." in {
    val root = new LC124_BinaryTreeMaximumPathSum.TreeNode(2)
    root.left = new LC124_BinaryTreeMaximumPathSum.TreeNode(1)
    root.right = new LC124_BinaryTreeMaximumPathSum.TreeNode(3)

    assert(LC124_BinaryTreeMaximumPathSum.maxPathSum(root) === 6)
  }
}
